store/models/job_spec.go

```diff
@@ -236,17 +244,43 @@ func (t TaskSpec) MarshalJSON() ([]byte, error) {
 	return json.Marshal(merged)
 }
 
+type taskType struct {
+	string
```

Comment:
Rather than embed string, I would create a new type. We have already done this with `Cron`:

```
type Cron string
```

https://github.com/smartcontractkit/chainlink/blob/master/store/models/common.go#L298-L300

### Why?
I tried finding a definitive resource to support my suggestion and couldn't really find one. It does seem subjective and arbitrary. Would love to hear or read some opinions on the matter of custom types vs embeddings for simple types.

So this suggestion is optional.

Reply:
There are some subtle differences, we did start this way, but it didn't work.